packages/host/src/components/Initialise/hooks.ts

```diff
@@ -0,0 +1,130 @@
+import { useEffect, useState } from 'react';
+import create from 'zustand';
+import { AppRoute } from '@openmsupply-client/config';
+import {
+  AuthenticationError,
+  AuthError,
+  LocalStorage,
+  ServerStatus,
+  useNavigate,
+} from '@openmsupply-client/common';
+import { useHost } from '../../api/hooks';
+
+const SERVER_RESTART_TIMEOUT = 90000;
+const POLLING_INTERVAL = 3000;
+
+interface InitialiseForm {
+  error?: AuthenticationError;
+  isLoading: boolean;
+  password: string;
+  siteId?: number;
+  username: string;
+  url: string;
+  setError: (error?: AuthenticationError) => void;
+  setIsLoading: (isLoading: boolean) => void;
+  setPassword: (password: string) => void;
+  setUsername: (username: string) => void;
+  setSiteId: (siteId: number) => void;
+  setUrl: (url: string) => void;
+}
+
+const useInitialiseFormState = create<InitialiseForm>(set => ({
+  error: undefined,
+  isLoading: false,
+  password: '',
+  username: '',
+  url: '',
+  siteId: undefined,
+  setError: (error?: AuthenticationError) =>
+    set(state => ({ ...state, error })),
+  setIsLoading: (isLoading: boolean) => set(state => ({ ...state, isLoading })),
+  setPassword: (password: string) => set(state => ({ ...state, password })),
+  setUsername: (username: string) => set(state => ({ ...state, username })),
+  setUrl: (url: string) => set(state => ({ ...state, url })),
+  setSiteId: (siteId: number) => set(state => ({ ...state, siteId })),
+}));
+
+export const useInitialiseForm = () => {
+  const state = useInitialiseFormState();
+  const navigate = useNavigate();
+  const { mutateAsync: restart } = useHost.utils.restart();
+  const {
+    setIsLoading,
+    password,
+    setPassword,
+    username,
+    error,
+    setError,
+    url,
+    siteId,
+  } = state;
+  const [isPolling, setIsPolling] = useState(false);
+  const [isBootstrap, setIsBootstrap] = useState(false);
+  const { mutateAsync: update } = useHost.sync.update();
+  const { data } = useHost.utils.settings({
+    refetchInterval: POLLING_INTERVAL,
```

Comment:
the `refetchInterval` sets the query to re-query regularly. This way I can check to see if the server is responding with a configured and running response or not.
The `enabled` flag does what it sounds like. Have set that to a state value, so that it can be enabled after the server is updated and restarted.